health-services/project-factory/src/server/utils/onGoingCampaignUpdateUtils.ts

```
@@ -0,0 +1,806 @@
+    }
+    // Update the projectId in the request's boundaryProjectMapping
+    request.body.boundaryProjectMapping[boundary].projectId = entry.id;
+  });
+
+  return boundarySet;
+}
+
+
+async function fetchProjectFacilityWithProjectId(request: any, projectId: any, facilityId: any) {
+  try {
+    const { tenantId } = request?.body?.parentCampaign || request?.parentCampaign;
```

Biome

[error] Unsafe usage of optional chaining.

If it short-circuits with 'undefined' the evaluation will throw TypeError here:


(lint/correctness/noUnsafeOptionalChaining)